ui/uihandler.go

```diff
@@ -386,10 +394,99 @@ func LogAndAddCacheControl(h http.Handler) http.Handler {
 	})
 }
 
-// LogDataRequest logs the data request.
-func LogDataRequest(h http.Handler) http.Handler {
+func sendHTMLDataIndex(w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "text/html; charset=UTF-8")
+	w.Write([]byte("<html><body><ul>\n")) // nolint: errcheck, gas
+	for _, e := range GetDataList() {
+		w.Write([]byte("<li><a href=\"")) // nolint: errcheck, gas
+		w.Write([]byte(e))                // nolint: errcheck, gas
+		w.Write([]byte(".json\">"))       // nolint: errcheck, gas
+		w.Write([]byte(e))                // nolint: errcheck, gas
+		w.Write([]byte("</a>\n"))         // nolint: errcheck, gas
+	}
+	w.Write([]byte("</ul></body></html>")) // nolint: errcheck, gas
+}
+
+type tsvCache struct {
+	cachedDirTime time.Time
+	cachedResult  []byte
+}
+
+var (
+	gTsvCache      tsvCache
+	gTsvCacheMutex = &sync.Mutex{}
+)
+
+// format for gcloud transfer
+// https://cloud.google.com/storage/transfer/create-url-list
+func sendTsvDataIndex(urlPrefix string, w http.ResponseWriter) {
```

Comment:
should this be: `sendTSVDataIndex` ?

I think TSV should be all-caps everywhere.